src/pages/VenuePage/VenuePage.tsx

```diff
@@ -115,77 +93,47 @@ const VenuePage: React.FC = () => {
     currentTimestamp >
       event.start_utc_seconds + event.duration_minutes * ONE_MINUTE_IN_SECONDS;
 
-  const isUserVenueOwner = user && venue?.owners?.includes(user.uid);
+  const isUserVenueOwner = userId && venue?.owners?.includes(userId);
   const isMember =
     user && venue && isUserAMember(user.email, venue.config?.memberEmails);
 
-  // Camp and PartyMap needs to be able to modify this
-  // Currently does not work with roome
-  const location = venueName;
-  useLocationUpdateEffect(user, venueName);
+  const updateUserLocation = useCallback(() => {
+    if (!userId || !venueName) return;
 
-  const newLocation = { [location]: getCurrentTimeInMilliseconds() };
-  const isNewLocation = profile?.lastSeenIn
-    ? !profile?.lastSeenIn[location]
-    : false;
+    updateLocationData(userId, {
+      [venueName]: getCurrentTimeInMilliseconds(),
+    });
+  }, [user, venueName]);
 
-  const newLocations = {
-    ...prevLocations,
-    ...newLocation,
-  };
+  useInterval(() => {
+    updateUserLocation();
+  }, LOC_UPDATE_FREQ_MS);
+
+  useUpdateTimespentPeriodically(user, venueName);
 
   useEffect(() => {
-    if (
-      user &&
-      location &&
-      isNewLocation &&
-      ((!unmounted && !retainAttendance) || retainAttendance) &&
-      (!profile?.lastSeenIn || !profile?.lastSeenIn[location])
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-      setUnmounted(false);
-    }
-    if (
-      user &&
-      profile &&
-      (profile.lastSeenIn === null || profile?.lastSeenIn === undefined)
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-    }
-  }, [
-    isNewLocation,
-    location,
-    newLocation,
-    newLocations,
-    prevLocations,
-    profile,
-    retainAttendance,
-    unmounted,
-    user,
-  ]);
+    if (!userId || !venueName) return;
+
+    updateLocationData(userId, {
+      [venueName]: getCurrentTimeInMilliseconds(),
+    });
```

Comment:
It would be good to have a function that removes this 'implementation detail' of having to lay out the location data structure, and just be callable as something like: `updateUserLocation(venueName)`, `updateUserLocation(venueName, roomName)`, etc